From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Tue, 25 Aug 2020 20:32:32 +0200
Subject: Bug 1661094: Always define PKGCONF_REQUIRES_PRIVATE

When configuring without system NSPR the configuration variable
PKGCONF_REQUIRES_PRIVATE isn't ever set, leading to a .pc file that still
contains the @PKGCONF_REQUIRES_PRIVATE@ stub.

Ensure that we always define PKGCONF_REQUIRES_PRIVATE, by setting it to an
empty string in case no system-nsrp is enabled.
In this way, the pkg-config file stub will be always replaced.

Differential Revision: https://phabricator.services.mozilla.com/D88179

Bug-Mozilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1661094
Forwarded: yes
---
 build/moz.configure/nspr.configure | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/build/moz.configure/nspr.configure b/build/moz.configure/nspr.configure
index c103e44..4ec6f0c 100644
--- a/build/moz.configure/nspr.configure
+++ b/build/moz.configure/nspr.configure
@@ -61,8 +61,10 @@ def nspr_pkg(nspr_pkg):
         lib_dir=lib_dir,
     )
 
-@depends(nspr_minver, when='--with-system-nspr')
-def pkgconf_requires_private(nspr_minver):
+@depends('--with-system-nspr', nspr_minver)
+def pkgconf_requires_private(system_nspr, nspr_minver):
+    if not system_nspr:
+        return ''
     return 'Requires.private: %s' % nspr_minver
 
 set_config('PKGCONF_REQUIRES_PRIVATE', pkgconf_requires_private)
